src/Symfony/Component/Translation/MessageCatalogueProvider/ResourceMessageCatalogueProvider.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\MessageCatalogueProvider;
+
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Translator;
+
+/**
+ * MessageCatalogueProvider loads catalogue from resources.
+ *
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class ResourceMessageCatalogueProvider implements MessageCatalogueProviderInterface
```

Comment:
What about just `MessageCatalogueProvider` as this is really the default implementation that almost everyone is going to use, right?